src/main/java/se/diabol/jenkins/pipeline/PipelinePagination.java

```diff
@@ -0,0 +1,118 @@
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+Delivery Pipeline Plugin is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with Delivery Pipeline Plugin.
+If not, see <http://www.gnu.org/licenses/>.
+*/
+package se.diabol.jenkins.pipeline;
+
+import hudson.model.Hudson;
+
+
+public class PipelinePagination {
+	private int totalCount = 0;
+	private int currentPage = 1;
+	private int prevPage = 0;
+	private int nextPage = 0;
+	private int totalPage = 0;
+	private int count = 0;
+	private int pageSize = 10;
+	private final String moveLink;
+	private String rootUrl = "";
+
+	
+	public PipelinePagination(int currentPage, int totalCount, int pagingSize, String moveLink) {
+		this.currentPage = currentPage;
+		this.totalCount = totalCount;
+		this.pageSize = pagingSize == 0 ? 10 : pagingSize;
+		this.moveLink = moveLink;
+		try {
+			rootUrl = Hudson.getInstance().getRootUrlFromRequest();
+		} catch (Exception e) {
+
+		}
+	}
+
+	
+	public String getTag() {		
+		setPageNation();
+		StringBuilder sb = new StringBuilder();
+		//sb.append("<table summary='page navigation' class='Nnavi'>\n").append("<tr>\n");
+		if (prevPage + 1 > 10) {
+			setPrePage(sb);
+		} else {
+			//setBlankSpan(sb);
+		}
+		setIndex(sb);
+		if (totalPage > (prevPage + 10)) {
+			setNextPage(sb);
+		}
+		//sb.append("</tr>\n").append("</table>\n");
+		return sb.toString();
+	}
+
+	private void setNextPage(StringBuilder sb) {
+		sb.append("<a href='").append(moveLink)
+				.append(nextPage).append("'>\n").append("Next").append("<img src='").append(rootUrl)
+				.append("/plugin/delivery-pipeline-plugin/nextbutton.gif' width='3' height='5' alt='' border='0'>\n")
+				.append("</a>\n");		
+	}
+
+	private void setPrePage(StringBuilder sb) {
+		sb.append("<a href='").append(moveLink).append(prevPage).append("'>\n")
+				.append("<img src='").append(rootUrl)
+				.append("/plugin/delivery-pipeline-plugin/prevbutton.gif' width='3' height='5' alt='' border='0'>\n")
+				.append("Prev").append("</a>\n");		
+	}
+
+	private void setBlankSpan(StringBuilder sb) {
+		sb.append("<span class='page dark'>\n");
+		sb.append("</span>");
+	}
+
+	private void setIndex(StringBuilder sb) {
+		for (count = prevPage + 1; count < nextPage && count <= totalPage; count++) {			
+			if (count == currentPage) {
+				sb.append("<span");
+				sb.append(" class='active_link'>\n");
+				sb.append("<a>").append(count).append("</a>\n");
+				sb.append("</span>\n");
+			} else {				
+				sb.append("<a href='").append(moveLink).append(count).append("'>").append(count).append("</a>\n");
+			}
+			
+		}
+
+	}
+
+	private void setPageNation() {
+		int current = (currentPage - 1) / 10 + 1;
```

Comment:
I think the intendation of the file is wrong, there seem to be too many spaces added when the code is intended. We don't currently have any style guidelines I'm afraid but a good start is to use the standard Java convention